packages/workspace/src/browser/workspace-storage-service.ts

```diff
@@ -21,18 +22,25 @@ export class WorkspaceStorageService implements StorageService {
     private initialized: Promise<void>;
     protected storageService: StorageService;
 
-    constructor( @inject(WorkspaceService) protected workspaceService: WorkspaceService,
-        @inject(ILogger) protected logger: ILogger) {
-        this.initialized = this.workspaceService.root.then(stat => {
-            if (stat) {
-                this.prefix = stat.uri;
-            } else {
-                this.prefix = '_global_';
-            }
-        });
+    @inject(WorkspaceService)
+    protected workspaceService: WorkspaceService;
+
+    @inject(ILogger)
+    protected logger: ILogger;
+
+    constructor() {
         this.storageService = new LocalStorageService(this.logger);
     }
 
+    @postConstruct()
+    protected async init(): Promise<void> {
+        const domain = await this.workspaceService.getDomain();
+        const statFile = await this.workspaceService.root;
+        const workspace = statFile ? new URI(statFile.uri).path : '_global_';
+        this.prefix = `${domain ? domain : window.location.pathname}:${workspace}`;
```

Comment:
You again add the pathname, only, and renamed the previous workspace id to domain? 
Can't you please just solve your issue in an extension that is specific for your workspace server?

Reply:
Would just adding `window.location.pathname` be enough ?

Reply:
That would be fine with me.